### PR TITLE
Bug 1871548 - Add telemetry for Open PDF in Fenix

### PR DESCRIPTION
**Pull request by titooan**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1871548

### CODE REVIEW

**Comment by titooan** [comment]
# Request for data collection review form

**All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.**

1) What questions will you answer with this data?
How many users open Fenix from an other app to view a PDF

2) Why does Mozilla need to answer these questions?  Are there benefits for users? Do we need this information to address product or business requirements?
We need to understand how much value this feature can bring to the users, if we need to improve this feature's visibility for the users, and how this feature impacts DAU.

3) What alternative methods did you consider to answer these questions? Why were they not sufficient?
There are currently no other alternatives.

4) Can current instrumentation answer these questions?
No

5) List all proposed measurements and indicate the category of data collection for each measurement, using the [Firefox data collection categories](https://wiki.mozilla.org/Data_Collection) found on the Mozilla wiki.

**Note that the data steward reviewing your request will characterize your data collection based on the highest (and most sensitive) category.**

<table>
  <tr>
    <td>Measurement Description</td>
    <td>Data Collection Category</td>
    <td>Tracking Bug #</td>
  </tr>
  <tr>
    <td>If the PDF was opened from Fenix itself (for example from the Download notification)</td>
    <td>Category 2 - interaction data</td>
    <td>[Bug 1871548](https://bugzilla.mozilla.org/show_bug.cgi?id=1871548)</td>
  </tr>
</table>

6) Please provide a link to the documentation for this data collection which describes the ultimate data set in a public, complete, and accurate way.
This collection is documented in the Glean Dictionary at https://dictionary.telemetry.mozilla.org/

7) How long will this data be collected?  Choose one of the following:
I want to permanently monitor this data. (put someone's name here -> @calixte? @sylvestre?)

8) What populations will you measure?
All channels, all locales, all countries

9) If this data collection is default on, what is the opt-out mechanism for users?
Default Glean SDK opt-out mechanism.

10) Please provide a general description of how you will analyze this data.
Glean.

11) Where do you intend to share the results of your analysis?
Only on Glean and with mobile teams.

12) Is there a third-party tool (i.e. not Glean or Telemetry) that you are proposing to use for this data collection?
No third-party tool

File: fenix/app/metrics.yaml [hunk 1]
```diff
@@ -376,6 +376,24 @@ events:
...
         - PrivateBrowsing
+  opened_ext_pdf:
+    type: event
+    description: |
+      A user opened a PDF with Fenix from another app
+    extra_keys:
+      referrer_is_fenix:
+        description: |
+          If the PDF was opened from Fenix itself (for example from the Download notification)
+        type: boolean
```
Comment (Amejia481) [inline fenix/app/metrics.yaml:387]: Should we also send an event when users click a PDF file on the "Downloads" screen?  :)

Comment (sylvestre) [reply fenix/app/metrics.yaml:387]: @Amejia481 maybe but I don't think it should be a blocker for this PR IMHO. It is less important

File: fenix/app/src/main/java/org/mozilla/fenix/IntentReceiverActivity.kt [hunk 2]
```diff
@@ -73,6 +73,12 @@ class IntentReceiverActivity : Activity() {
 
         addReferrerInformation(intent)
 
+        if (intent.type == "application/pdf") {
```
Comment (Mukuljangir372) [inline fenix/app/src/main/java/org/mozilla/fenix/IntentReceiverActivity.kt:76]: Should we move `application/pdf` to companion object as content?

File: fenix/app/metrics.yaml (hunk 1 above)
Comment (titooan) [reply fenix/app/metrics.yaml:387]: Actually with this patch we send "true" for both cases: opening from the Downloads screen and opening from the Donwload notification (in app, or in the Notification Tray). So I guess this is already covered

File: fenix/app/src/main/java/org/mozilla/fenix/IntentReceiverActivity.kt (hunk 2 above)
Comment (titooan) [reply fenix/app/src/main/java/org/mozilla/fenix/IntentReceiverActivity.kt]: I forgot that, good catch, thanks! It's fixed now

File: fenix/app/metrics.yaml (hunk 1 above)
Comment (Amejia481) [reply fenix/app/metrics.yaml:387]: Perfect, from my side it looks good we just need the data steward approval too :)

Comment (Amejia481) [reply fenix/app/metrics.yaml:387]: @titooan I'm not sure but product maybe interested into identify how many PDF viewer requests are coming from our app or other apps.

**Review by Amejia481 — APPROVED** [review]
LGTM!

**Comment by rocketsroger** [comment]
> # Request for data collection review form
> 
> **All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.**
> 
>     1. What questions will you answer with this data?
> 
> 
>     * How many users open Fenix from an other app to view a PDF
> 
> 
>     2. Why does Mozilla need to answer these questions?  Are there benefits for users? Do we need this information to address product or business requirements?
> 
> 
>     * We need to understand how much value this feature can bring to the users, if we need to improve this feature's visibility for the users, and how this feature impacts DAU.
> 
> 
>     3. What alternative methods did you consider to answer these questions? Why were they not sufficient?
> 
> 
>     * There are currently no other alternatives.
> 
> 
>     4. Can current instrumentation answer these questions?
> 
> 
>     * No
> 
> 
>     5. List all proposed measurements and indicate the category of data collection for each measurement, using the [Firefox data collection categories](https://wiki.mozilla.org/Data_Collection) found on the Mozilla wiki.
> 
> 
> Measurement Description 	Data Collection Category 	Tracking Bug #
> If the PDF was opened from Fenix itself (for example from the Download notification) 	Category 2 - interaction data 	[Bug 1871548](https://bugzilla.mozilla.org/show_bug.cgi?id=1871548)
> 
>     6. Please provide a link to the documentation for this data collection which describes the ultimate data set in a public, complete, and accurate way.
> 
> 
>     * This collection is documented in the Glean Dictionary at https://dictionary.telemetry.mozilla.org/
> 
> 
>     7. How long will this data be collected?  Choose one of the following:
> 
> 
>     * I want to permanently monitor this data.
> 
> 
>     8. What populations will you measure?
> 
> 
>     * All channels, all locales, all countries
> 
> 
>     9. If this data collection is default on, what is the opt-out mechanism for users?
> 
> 
>     * Default Glean SDK opt-out mechanism.
> 
> 
>     10. Please provide a general description of how you will analyze this data.
> 
> 
>     * Glean.
> 
> 
>     11. Where do you intend to share the results of your analysis?
> 
> 
>     * Only on Glean and with mobile teams.
> 
> 
>     12. Is there a third-party tool (i.e. not Glean or Telemetry) that you are proposing to use for this data collection?
> 
> 
>     * No third-party tool

# Data Review

1) Is there or will there be **documentation** that describes the schema for the ultimate data set in a public, complete, and accurate way? 

Yes, through the metrics.yaml file and the [Glean Dictionary](https://dictionary.telemetry.mozilla.org/apps/fenix )

2) Is there a control mechanism that allows the user to turn the data collection on and off? 

Yes, through the "Send Usage Data" preference in the application settings

3) If the request is for permanent data collection, is there someone who will monitor the data over time?

mcastelluccio@mozilla.com, calixte@mozilla.com and sylvestre@mozilla.com will be responsible for monitoring the data over time

4) Using the **[category system of data types](https://wiki.mozilla.org/Data_Collection)** on the Mozilla wiki, what collection type of data do the requested measurements fall under?

Category 2, Interaction data

5) Is the data collection request for default-on or default-off?

default-on

6) Does the instrumentation include the addition of **any *new* identifiers**?

No

7) Is the data collection covered by the existing Firefox privacy notice? 

Yes

8) Does the data collection use a third-party collection tool? 

No

## Result

data-review+

File: fenix/app/metrics.yaml [hunk 3]
```diff
@@ -376,6 +376,27 @@ events:
...
+    bugs:
+      - https://bugzilla.mozilla.org/show_bug.cgi?id=1871548
+    data_reviews:
+      - TODO
```
Comment (rocketsroger) [inline fenix/app/metrics.yaml:391]: Don't forget to update this to this PR's url.

**Comment by titooan** [comment]
@Mergifyio backport releases_v122

**Pull request merged**